MapleServer2/Types/Achieve.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using MapleServer2.Data.Static;
+
+namespace MapleServer2.Types
+{
+    public class Achieve
+    {
+        public Player Owner { get; private set; }
+        public int Id { get; private set; }
+        public int Grade {get; private set; }   // grade trying to achieve
+        public int NumGrades {get; private set; }
+        public long Counter { get; private set; }    // counter to reach next grade
+        public long Max { get; private set; }    // counter value needed to reach next grade (-1 if grade is max)
+        public List<long> Timestamps { get; private set; }  // stored in ascending order of grades as seconds since epoch time
+        
+        public Achieve(int achieveId, int grade = 0, int counter = 0, List<long> timestamps = null)
+        {
+            this.Id = achieveId;
```

Comment:
Remove `this` and let `Id = achieveId`, cause public fields ar PascalCase and parameters are camelCase, remove the others as well.